case-lib/opt.sh

```diff
@@ -1,141 +1,158 @@
+            [ "${OPT_HAS_ARG[$opt]}" -ne 1 ] || {
+                short_opt_str=$short_opt_str':'
+                long_opt_str=$long_opt_str':'
+            }
         done
-
-        if  ! _op_temp_script=$(
-                getopt -o "$_short_opt" --long "$_long_opt" -- "$@"); then
-            # here output for the End-User who don't care about this function/option couldn't to run
-            printf 'Error parsing options: %s\n' "$*"
-            # To debug uncomment these lines:
-            # printf 'DEBUG parsing options: short_opt: %s / long_opt: %s / args: %s\n' "$_short_opt" "$_long_opt" "$*" >&2
-            # declare -p |grep -E 'OPT_[A-Z]*_lst|_op_'
-            _func_opt_dump_help
-        fi
     }
 
-    _func_opt_dump_help()
+    _dump_help_and_exit()
     {
         local i
+        local exit_status="$1"
+
         printf 'Usage: %s [OPTION]\n' "$0"
         for i in ${!OPT_DESC[*]}
         do
-                [ "X$i" = "Xh" ] && continue
-                # display short option
-                [ "$i" ] && printf '    -%s' "$i"
-                # if option requires extra argument
-                [ "X${OPT_HAS_ARG[$i]}" == "X1" ] && printf ' parameter'
-                # display long option
-                if [ "${OPT_NAME[$i]}" ]; then
-                    # whether display short option
-                    [ "$i" ] && printf ' |  ' || printf '    '
-                    printf '%s' "--${OPT_NAME[$i]}"
-                    [ "X${OPT_HAS_ARG[$i]}" == "X1" ] && printf ' parameter'
-                fi
-                printf '\n\t%s\n' "${OPT_DESC[$i]}"
-                if [ "${OPT_VAL[$i]}" ]; then
-                    if [ "${OPT_HAS_ARG[$i]}" -eq 1 ]; then
-                        printf '\tDefault Value: %s\n' "${OPT_VAL[$i]}"
-                    elif [ "${OPT_VAL[$i]}" -eq 0 ]; then
-                        printf '\tDefault Value: Off\n'
-                    else
-                        printf '\tDefault Value: On\n'
-                    fi
+            [ "X$i" != "Xh" ] || continue
+            # display short option
+            [ ! "$i" ] || printf '    -%s' "$i"
+            # if option requires extra argument
+            [ "X${OPT_HAS_ARG[$i]}" != "X1" ] || printf ' parameter'
+            # display long option
+            if [ "${OPT_NAME[$i]}" ]; then
+                # whether display short option
+                [ "$i" ] && printf ' |  ' || printf '    '
+                printf '%s' "--${OPT_NAME[$i]}"
+                [ "X${OPT_HAS_ARG[$i]}" != "X1" ] || printf ' parameter'
+            fi
+            printf '\n\t%s\n' "${OPT_DESC[$i]}"
+            if [ "${OPT_VAL[$i]}" ]; then
+                if [ "${OPT_HAS_ARG[$i]}" -eq 1 ]; then
+                    printf '\tDefault Value: %s\n' "${OPT_VAL[$i]}"
+                elif [ "${OPT_VAL[$i]}" -eq 0 ]; then
+                    printf '\tDefault Value: Off\n'
+                else
+                    printf '\tDefault Value: On\n'
```

Comment:
Next time please don't mix in the same PR such a big whitespace change with actual code changes because it's practically impossible to see whether any code has changed there.